### DISCUSSION TITLE
High CPU usage when reading HLS and externalAuthenticationURLis enabled

### DISCUSSION THREAD

**Discussion by brian7704** [bluenviron/mediamtx#3003] [Questions]
### Question

I've noticed high CPU usage when a reader connects to an HLS stream and externalAuthenticationURLis enabled. Looking at the logs I see that auth happens every few seconds, which I assume means every time there is a new segment. Besides changing the segment duration, is there a way to reduce the number of times auth occurs? Or is there a way to make auth only occur when the reader first connects?

**Comment by aler9** [comment]
since v1.6.0 there are two options to solve the issue:

* use `authHTTPExclude` to allow reading streams without passing through the external authentication server:

   ```yml
   authHTTPExclude:
  - action: read
   ```

* use JWT-based authentication. Each reader has to request a JWT from an identity server (the README contains instructions on how to setup the Keycloak identity server); this JWT can be used to request HLS:

   ```
   http://localhost:8888/stream/?jwt=my_jwt
   ```

[a comment by an automated account is not shown]
